src/ngraph/op/roi_align.cpp

```diff
@@ -0,0 +1,153 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "roi_align.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::v0::ROIAlign::type_info;
+
+op::v0::ROIAlign::ROIAlign(const Output<Node>& input,
+                           const Output<Node>& rois,
+                           const Output<Node>& batch_indices,
+                           const size_t pooled_h,
+                           const size_t pooled_w,
+                           const size_t sampling_ratio,
+                           const float spatial_scale,
+                           const std::string& mode)
+    : Op{{input, rois, batch_indices}}
+    , m_pooled_h{pooled_h}
+    , m_pooled_w{pooled_w}
+    , m_sampling_ratio{sampling_ratio}
+    , m_spatial_scale{spatial_scale}
+    , m_mode{mode}
```

Comment:
I think that we should have second constructor which get mode as enum and use the enum version in `validate_and_infer_types`.
Similar solution was used for example to `TopK`: https://github.com/NervanaSystems/ngraph/blob/d0d63c304f9856c3097783d5e380e231d3629e57/src/ngraph/op/topk.cpp#L412